olp-cpp-sdk-core/include/olp/core/client/OlpClient.h

```diff
@@ -67,13 +71,19 @@ class CORE_API OlpClient {
   /**
    * @brief Gets the default headers that are added to each request.
    *
+   * @note Do not change this while requests are ongoing.
+   *
    * @return The default headers.
    */
   ParametersType& GetMutableDefaultHeaders();
 
   /**
    * @brief Sets the client settings.
    *
+   * @note Handle with care and do not change while requests are ongoing.
+   * Ideally the settings would not change during the lifecycle of this
+   * instance.
+   *
```

Comment:
Maybe we can deprecate this?

Reply:
Seems reasonable. Will do it in a separate PR.